### PR TITLE
141 UI for recipe step creation

### PR DESCRIPTION
**Pull request by fdejean**
Display a list of Step

- [x] Match the figma
- [x] Added a colapse feature
- [x] Added error warning for empty title and description

### CODE REVIEW

**Comment by 0marwa** [comment]
I'll leave a preview here for reference (the background is transparent for some reason?) 
Screenshot 2024-05-05 at 12 47 50

**Review by 0marwa — APPROVED** [review]
Good job having the description field shown when we expand the title, as the description is optional :) 
Other than that, nothing to signal!

File: app/src/main/java/com/android/feedme/ui/component/RecipeStepInput.kt [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+          },
+          onDeleteStep = {
+            // stepViewModel.deleteStep(index)
+          })
+    }
+  }
+}
+
+@Composable
+fun StepInput(step: Step, onStepChanged: (Step) -> Unit, onDeleteStep: () -> Unit) {
```
Comment (JeannedeM) [inline app/src/main/java/com/android/feedme/ui/component/RecipeStepInput.kt:41]: Hello, great UI ! Thank you ! Could you just add documentation for the function here please :)

**Review by JeannedeM — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/android/feedme/ui/component/RecipeStepInput.kt (hunk 1 above)
Comment (fdejean) [reply app/src/main/java/com/android/feedme/ui/component/RecipeStepInput.kt:41]: Hey, yes, I will add some! 
Thanks

**Comment by fdejean** [comment]
@0marwa @JeannedeM thanks for the reivew,
Request changed have been implemented,

Here is the UI with a white background
Note that the step number depend on the Step object, and since it use the same step, the step number does not change.
That behavior will change in #146 

Screenshot_20240505_144810_feedmeApp

**Review by JeannedeM — APPROVED** [review]

**Pull request merged**